board/tests.py

```diff
@@ -146,4 +149,178 @@ def test_post_creation_combinations(self):
         p2.save()
         p3.save()
         p4.save()
-        p5.save()
\ No newline at end of file
+        p5.save()
+
+
+class APITests(TestCase):
+    """Tests with API endpoints."""
+
+    def add_posts(self, board, amount):
+        """Return a list of posts that are added to a board."""
+        posts = []
+        for i in range(amount):
+            p = Post(
+                associated_board=board, 
+                message='hi', 
+                name=str(i), 
+                photo=Image(name=f'photo{i}', photo=bytearray('hi', 'utf-8')).save(),
+                created_at=timezone.now()+timezone.timedelta(seconds=i)
+            )
+            p.save()
+            posts.append(p)
+        return posts
+
+    @tag('core')
+    def test_get_posts(self):
+        """Make sure that it returns proper JSON responses with proper amount of posts."""
+        
+        b = Board(title='hi', description='hello', bg=Image(name='i', photo=bytearray('hi', 'utf-8')).save())
+        b.save()
+        posts = self.add_posts(b, 30)
+        
+        # First 10 posts
+        res1 = self.client.get(
+            reverse('board:posts-get'),
+            {
+                'board': str(b.uuid),
+                'index': '0',
+                'amount': '10',
+            },
+            HTTP_ACCEPT='application/json',
+        )
+        exp1 = []
+        for i in range(10):
+            p = posts[-i-1]
+            exp1.append(GetPosts.get_post_dict(p))
+        self.assertEqual(res1.json(), exp1)
+
+
+        # Second 10 posts
+        res2 = self.client.get(
+            reverse('board:posts-get'),
+            {
+                'board': str(b.uuid),
+                'index': '10',
+                'amount': '10',
+            },
+            HTTP_ACCEPT='application/json',
+        )
+        exp2 = []
+        for i in range(10):
+            p = posts[-i-11]
+            exp2.append(GetPosts.get_post_dict(p))
+        
+        self.assertEqual(res2.json(), exp2)
+
+    @tag('core')
+    def test_get_posts_fully_loaded(self):
+        """
```

Comment:
In Pydocs there should be a one-liner summary docstring right after the three quotes. eg:

```
"""Test for x.

It should return y.
"""
```

See https://www.python.org/dev/peps/pep-0257/ for all the recommendations

Reply:
newwwwwwwwwwwwww issuueeeeeeeeeeeeee 🤡